django/db/models/constraints.py

```diff
@@ -65,13 +65,16 @@ def deconstruct(self):
 
 
 class UniqueConstraint(BaseConstraint):
-    def __init__(self, *, fields, name, condition=None):
+    def __init__(self, *, fields, name, condition=None, include=()):
         if not fields:
             raise ValueError('At least one field is required to define a unique constraint.')
         if not isinstance(condition, (type(None), Q)):
             raise ValueError('UniqueConstraint.condition must be a Q instance.')
+        if include and not isinstance(include, (list, tuple)):
+            raise ValueError('UniqueConstraint.include must be a list or tuple.')
         self.fields = tuple(fields)
         self.condition = condition
+        self.include = tuple(include) if include else None
```

Comment:
Make this `tuple(include or ())` and then you don't need `[... in self.include or []]` in the list comprehensions below.